presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -528,6 +569,7 @@ private void queryFinished(QueryExecution query)
                     group = group.parent.orElse(null);
                 }
             }
+            activeQueries.remove(query);
```

Comment:
(will come back to this design decision, but if you want to go this route, you should also cleanup in line 556.